libs/api/domains/municipalities-financial-aid/src/lib/municipalitiesFinancialAid.resolver.ts

```diff
@@ -0,0 +1,69 @@
+import { Args, Mutation, Query, Resolver } from '@nestjs/graphql'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  IdsUserGuard,
+  ScopesGuard,
+  CurrentUser,
+} from '@island.is/auth-nest-tools'
+import { UseGuards } from '@nestjs/common'
+
+import { MunicipalitiesFinancialAidService } from './municipalitiesFinancialAid.service'
+import { MunicipalityModel } from './models/municipality.model'
+import { MunicipalityQueryInput } from './models/municipality.input'
+import { SignedUrlModel } from './models/signedUrl.model'
+import { GetSignedUrlInput } from './dto/getSignedUrl.input'
+import { CreateMunicipalitiesApplicationInput } from './dto/createApplication.input'
+import { ApplicationModel } from './models/application.model'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver()
+export class MunicipalitiesFinancialAidResolver {
+  constructor(
+    private municipalitiesFinancialAidService: MunicipalitiesFinancialAidService,
+  ) {}
+
+  @Query(() => String, { nullable: true })
+  async hasUserFinancialAidApplicationForCurrentPeriod(
```

Comment:
All of the queries, mutations and object types need to be [namespaced consistently](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api).

`MunicipalitiesFinancialAid` is a pretty long namespace but maybe fine. You can have a "big" `Municipalities` namespace, a smaller `MuniFA` or anything between. :)

Note that you can have a separate TypeScript name vs GraphQL name (passed in as a decorator argument).

Also please review the [overall graphql conventions](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions). E.g. skip `get` in `{namespace}SignedUrl`, verbs are generally allowed for mutations like this: `{verb}{Namespace}{Model}`.

Regarding the `has` query, I'd generally try to structure it using a `get` query with some input parameters eg `{namespace}Application({ period: 'current' })` . If that doesn't make sense then `has` is fine as a verb `has{Namespace}ApplicationForCurrentPeriod`